### PR TITLE
Consume sandbox with variable change tracking enabled

### PR DESCRIPTION
**Pull request by kamalaknn**
* Update to sandbox that now provides variable changes within the script execution
* Use the variable changes for internally updating variables

### CODE REVIEW

File: lib/runner/extensions/event.command.js [hunk 1]
```diff
@@ -288,7 +288,7 @@ module.exports = {
                 // finally execute the script
                 this.host.execute(event, {
                     id: executionId,
-                    // debug: true,
+                    debug: true,
```
Comment (kunagpal) [inline lib/runner/extensions/event.command.js]: What exactly does `debug` do?

File: lib/runner/extensions/event.command.js [hunk 2]
```diff
@@ -331,6 +336,7 @@ module.exports = {
 
                     // Ensure that we have SDK instances, not serialized plain objects.
                     // @todo - should this be handled by the sandbox?
+                    result && result._variables && (result._variables = new sdk.VariableScope(result._variables));
```
Comment (kunagpal) [inline lib/runner/extensions/event.command.js]: Shouldn't there be a stricter check on `result._variables?`

File: test/integration/sanity/variable-changes.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                            exec: `
+                                pm.variables.set('_variable', '_variable value');
+                                pm.environment.set('environment', 'environment value');
+                                pm.globals.set('globals', 'globals value');
+                            `
+                        }
+                    }, {
+                        listen: 'test',
+                        script: {
+                            exec: `
```
Comment (kunagpal) [inline test/integration/sanity/variable-changes.test.js]: Can be condensed into one line.

File: lib/runner/extensions/event.command.js [hunk 4]
```diff
@@ -308,19 +308,24 @@ module.exports = {
...
-                        // scope using that
-                        if (_.isFunction(payload.context[variable].syncVariablesFrom)) {
-                            payload.context[variable].clear();
-                            _.forEach(result[variable].values, function (def) {
-                                payload.context[variable].values.add(def);
-                            });
+                        var mutations,
+                            contextVariable = payload.context[variable];
+
+                        if (!(_.isObject(result[variable]) && result[variable].mutations && contextVariable)) {
```
Comment (kunagpal) [inline lib/runner/extensions/event.command.js]: The `if` block has to be shifted to before the `var ` above, purely for the reasons of readability.

**Comment by kunagpal** [comment]
@kamalaknn Also, builds are failing.

**Comment by kamalaknn** [comment]
@kunagpal I haven't committed the dependency bump on sandbox. I'll do it once we have a release.

File: lib/runner/extensions/event.command.js (hunk 1 above)
Comment (kamalaknn) [reply lib/runner/extensions/event.command.js]: shouldn't have been committed. Good catch.

File: lib/runner/extensions/event.command.js (hunk 2 above)
Comment (kamalaknn) [reply lib/runner/extensions/event.command.js]: VariableScope constructor handles weirdness in the parameters.

File: lib/runner/extensions/event.command.js [hunk 5]
```diff
@@ -308,19 +308,25 @@ module.exports = {
 
                     // if it is defined that certain variables are to be synced back to result, we do the same
                     track && result && track.forEach(function (variable) {
-                        if (!(_.isObject(result[variable]) && payload.context[variable])) { return; }
```
Comment (shamasis) [inline lib/runner/extensions/event.command.js]: keep this

**Pull request merged**